File: 프로그래머스/Level_1/pro_81301.js
// Lv.1 - 숫자 문자열과 영단어

function solution(s) {
    var answer = "";
    const number = ["zero", "one","two","three","four","five","six","seven","eight","nine"];
    let word = "";

    for(let i = 0; i < s.length; i++){
        if(isNaN(parseInt(s[i]))){
            word += s[i];
        }
        else{
            answer += s[i];
        }
        if(number.includes(word)){
            answer += number.indexOf(word);
            word = "";
        }
    }

    answer = parseInt(answer);
    return answer;
}

const s1 = "one4seveneight";
const s2 = "23four5six7";
const s3 = "2three45sixseven"
const s4 = "123"

console.log(solution(s1));
console.log(solution(s2));
console.log(solution(s3));
console.log(solution(s4));

File: 프로그래머스/Level_1/pro_81301.java
// Lv.1 - 숫자 문자열과 영단어

import java.util.HashMap;

public class pro_81301 {
    public int solution(String s) {
        int answer = 0;
        String[] engWord = {"zero", "one", "two", "three","four","five","six","seven","eight","nine"};
        String a = "";
        String word = "";
        char ch;
        
        HashMap<String, String> map = new HashMap<String, String>();
        for(int i = 0; i < 10; i++){
            map.put(engWord[i], Integer.toString(i));
        }
        
        for(int i = 0; i < s.length(); i++){
            ch = s.charAt(i);
            if(ch >= 48 && ch <= 57){
                a += ch;
            }
            else{
                word += ch;
            }
            if(map.containsKey(word)){
                a += map.get(word); 
                word = "";
            }
        }
        
        answer = Integer.parseInt(a);
        return answer;
    }

    public static void main(String[] args){
        pro_81301 ne = new pro_81301();

        System.out.println(ne.solution("one4seveneight"));
        System.out.println(ne.solution("23four5six7"));
        System.out.println(ne.solution("2three45sixseven"));
        System.out.println(ne.solution("123"));
    }
}
